openapi/src/main/java/io/helidon/openapi/Serializer.java

```diff
@@ -0,0 +1,349 @@
+             * See https://github.com/eclipse/microprofile-open-api/issues/412
+             */
+            if (Number.class.isInstance(propertyValue) && !Boolean.getBoolean("io.helidon.openapi.skipTCKWorkaround")) {
+                Number n = (Number) propertyValue;
+                float diff = n.floatValue() - n.intValue();
+                if (diff == 0) {
+                    propertyValue = Integer.valueOf(n.intValue());
+                } else if (Math.abs(diff) < 0.1) {
+                    LOGGER.warning(String.format("Integer approximation of %f did not match but the difference was only %e",
+                            n, diff));
+                }
+            }
+            return propertyValue;
+        }
+
+        @Override
+        protected MappingNode representJavaBean(Set<Property> properties, Object javaBean) {
+            MappingNode result = super.representJavaBean(properties, javaBean);
+            processExtensions(result, javaBean);
+            representedObjects.clear();
+            return result;
+        }
+
+        private void processExtensions(MappingNode node, Object javaBean) {
+            if (!Extensible.class.isAssignableFrom(javaBean.getClass())) {
+                return;
+            }
+
+            List<NodeTuple> tuples = new ArrayList<>(node.getValue());
+
+            if (tuples.isEmpty()) {
+                return;
+            }
+            List<NodeTuple> updatedTuples = new ArrayList<>();
+
+            tuples.forEach(tuple -> {
+                Node keyNode = tuple.getKeyNode();
+                if (keyNode.getTag().equals(Tag.STR)) {
+                    String key = ((ScalarNode) keyNode).getValue();
+                    if (key.equals(EXTENSIONS)) {
+                        Node valueNode = tuple.getValueNode();
+                        if (valueNode.getNodeId().equals(NodeId.mapping)) {
+                            MappingNode extensions = MappingNode.class.cast(valueNode);
+                            updatedTuples.addAll(extensions.getValue());
+                        }
+                    } else {
+                        updatedTuples.add(tuple);
+                    }
+                } else {
+                    updatedTuples.add(tuple);
+                }
+            });
+            node.setValue(updatedTuples);
+        }
+
+        /**
+         * Some SmallRye implementation classes have properties not supported by the implemented interface, so ignore those or
+         * the serialized YAML will contain SmallRye-only properties.
+         *
+         * @param javaBean the bean being serialized
+         * @param property the property being serialized
+         * @return true if the property should be processes; false otherwise
+         */
+        private boolean okToProcess(Object javaBean, Property property) {
+            boolean reject = false;
+            reject |= Parameter.class.isAssignableFrom(javaBean.getClass()) && property.getName().equals("hidden");
```

Comment:
Been a while since I've seen this operator.  Won't this always assign `false` to `reject`?  {Heads off to JLS page}

Reply:
Oh, never mind.

Reply:
I'm adding a comment in the code explaining. If, over time, we have to add more cases to ignore then we can just add a new line similar to the existing `reject |= ...` one for each case to be skipped without perturbing the other lines in the method.